File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/operation/mutate/RecordsMutateDao.java
package ru.citeck.ecos.records3.record.operation.mutate;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import ru.citeck.ecos.records3.RecordAtts;
import ru.citeck.ecos.records3.RecordRef;
import ru.citeck.ecos.records3.source.dao.RecordsDao;

import java.util.List;

public interface RecordsMutateDao extends RecordsDao {

    @Nullable
    List<RecordRef> mutate(@NotNull List<RecordAtts> records);
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/dao/mutate/RecordsMutateDao.kt
package ru.citeck.ecos.records3.record.dao.mutate

import ru.citeck.ecos.records3.record.atts.dto.LocalRecordAtts
import ru.citeck.ecos.records3.record.dao.RecordsDao
import kotlin.jvm.Throws

interface RecordsMutateDao : RecordsDao {

    @Throws(Exception::class)
    fun mutate(records: List<LocalRecordAtts>): List<String>
}
